spec-dtslint/operators/count-spec.ts

```diff
@@ -0,0 +1,26 @@
+import { of } from 'rxjs';
+import { count, buffer } from 'rxjs/operators';
+
+it('should always infer number', () => {
+  const o = of(1, 2, 3).pipe(count(x => x > 1)); // $ExpectType Observable<number>
```

Comment:
With `count`, the return type of the piped observable will always be `Observable<number>`, so I think that for the tests, the source observable type should be something other than `number` - to make sure that the output type comes from the `count` operator and not the source. E.g.:

```ts
const o = of('1', '2', '3').pipe(count(x => x !== '1')); // $ExpectType Observable<number>
```

Reply:
@cartant I have added another row to both number or string will always return number